src/Uno.UI/UI/Xaml/Application.cs

```diff
@@ -91,6 +92,43 @@ public ApplicationTheme RequestedTheme
 			}
 		}
 
+		private ApplicationTheme? InternalRequestedTheme
+		{
+			get => _requestedTheme;
+			set
+			{
+				_requestedTheme = value;
+				UpdateRequestedThemesForResources();
+			}
+		}
+
+		internal static void UpdateRequestedThemesForResources()
+		{
+			string Update()
+			{
+				var custom = ApplicationHelper.RequestedCustomTheme;
+
+				if (!custom.IsNullOrEmpty())
+				{
+					return custom;
+				}
+
+				switch (Application.Current.RequestedTheme)
+				{
+					case ApplicationTheme.Light:
+						return "Light";
+					case ApplicationTheme.Dark:
+						return "Dark";
+					default:
+						throw new InvalidOperationException($"Theme {Application.Current.RequestedTheme} is not valid");
+				}
+			}
+
+			Application.Current.RequestedThemeForResources = Update();

```

Comment:
Why do you create a nested method you invoke immediately? Sounds less performant ...